api/src/main/java/net/hypejet/jet/plugin/PluginMetadata.java

```diff
@@ -0,0 +1,19 @@
+package net.hypejet.jet.plugin;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.util.Collection;
+
+/**
+ * Represents a metadata of a {@linkplain Plugin plugin}.
+ *
+ * @param name a name of the plugin
+ * @param version a version of the plugin
+ * @param mainClass a full name of the main class of the plugin
+ * @param authors authors of the plugin
+ * @param dependencies dependencies of the plugin
+ * @since 1.0
+ * @author Codestech
+ */
+public record PluginMetadata(@NonNull String name, @NonNull String version, @NonNull String mainClass,
```

Comment:
mainClass should probably be replaced with a hashmap of <String (name), String (path)>, like in fabric, this would make it easier to create addons for other plugins since plugins could check the pluginmanager for entrypoints maching their plugin

"main" would be the default entrypoint used by jet

[see this](https://fabricmc.net/wiki/documentation:entrypoint)